### ISSUE TITLE
Using this config on macOS doesn't correctly load the shell.

### ISSUE THREAD

**Issue by CrazyCatZhang** [KevinSilvester/wezterm-config#12]
I'm on macOS version 14.3.1, and I'm using the system's default zsh.
<img width="799" alt="image" src="https://github.com/KevinSilvester/wezterm-config/assets/67346540/e59c2b81-bde1-4888-8e95-93fbe7be606b">
The commands configured in the zsh environment variables are no longer working properly.

```
elseif platform.is_mac then
   options.default_prog = { '/bin/zsh' }
   options.launch_menu = {
      { label = 'Bash', args = { 'bash' } },
      { label = 'Fish', args = { '/opt/homebrew/bin/fish' } },
      { label = 'Nushell', args = { '/opt/homebrew/bin/nu' } },
      { label = 'Zsh', args = { 'zsh' } },
   }
```

**Comment by CrazyCatZhang** [comment]
I have already installed them.

**Comment by CrazyCatZhang** [comment]
I'm experiencing 'not found' errors for many commands that I've already installed. even brew is showing 'not found'.

**Comment by CrazyCatZhang** [comment]
solved.

**Comment by maxz255** [comment]
> I'm experiencing 'not found' errors for many commands that I've already installed. even brew is showing 'not found'.

I have the same issue, could you please tell me how you solved it?

**Comment by CrazyCatZhang** [comment]
<img width="1762" alt="image" src="https://github.com/KevinSilvester/wezterm-config/assets/67346540/8ad4cbbd-efc7-4278-a795-20056a9de64d">

like this, remove `options.default_prog`

**Comment by CrazyCatZhang** [comment]
> > I'm experiencing 'not found' errors for many commands that I've already installed. even brew is showing 'not found'.
> 
> I have the same issue, could you please tell me how you solved it?

<img width="1762" alt="image" src="https://github.com/KevinSilvester/wezterm-config/assets/67346540/8ad4cbbd-efc7-4278-a795-20056a9de64d">

like this, remove `options.default_prog`

**Comment by maxz255** [comment]
thx, solved it !

**Comment by KevinSilvester** [comment]
Hey guys, 
I think the root cause of these issues is a mistake on my part and how I defined the launch args for shells on UNIX systems. 😓

On UNIX systems, the `-l` flag is required to tell the shell to execute the system-wide + user-specific login scripts (e.g. `~/.bashrc`, `~/.profile`, `/etc/bashrc`, `/etc/profile`) before starting the shell.

This may have been why some programs (go, starship) were missing from the `$PATH` when launching with the default shell.

I've updated the `launch_menu` and `default_prog` options for Linux and Mac to include the `login` flag.
Hopefully, that will let you use your preferred shell without any issues.

<https://github.com/KevinSilvester/wezterm-config/blob/ff10cc0ba82ce7d90f84acfe66b2bf6da8921b8d/config/launch.lua#L1-L35>
